### PR TITLE
fix: icon textures and values did not update after state changes

### PR DESCRIPTION
**Pull request by ultijlam**
This pullup fixes issue #749.

### CODE REVIEW

**Comment by ultijlam** [comment]
Oof, I clearly forgot to change leading tabs into spaces. I'll make one more commit to fix that in this branch.

**Comment by ultijlam** [comment]
I think it was just my mobile phone not showing the edits properly. All leading tabs are already changed to 4 spaces. This is ready for pull-up.

**Comment by ultijlam** [comment]
I think optimizations can return if we properly set the age on all results. I think right now the age is not always updated, so you don't know when you can safely use an optimization or if a result is out-of-date.

File: src/engine/runner.ts [hunk 1]
```diff
@@ -185,31 +206,27 @@ export class Runner {
         atTime: number
     ): AstNodeSnapshot {
         this.profiler.StartProfiling("OvaleBestAction_RecursiveCompute");
-        if (element.result.constant) {
```
Comment (Sidoine) [inline src/engine/runner.ts:188]: Why did you remove this code?

Comment (ultijlam) [reply src/engine/runner.ts:188]: What has happening was an "if" node with an existing constant result would always return the constant result, even if the condition of the "if" node changed values due to, e.g., checkbox control changing, or items being equipped, etc.

I believe it's because the results aren't being aged correctly, because when I checked the ages of the results, the constant results had very low ages, while the "if" node itself had a very high age.

Comment (ultijlam) [reply src/engine/runner.ts:188]: I removed the constant optimization for correctness for now, and then I planned to see how to reintroduce it again.

Comment (ultijlam) [reply src/engine/runner.ts:188]: I'm planning on adding some tests to verify the ages of nodes after a script is evaluated.

Comment (Sidoine) [reply src/engine/runner.ts:188]: Constant nodes does not care of the age, it's for strings or numbers that should never change. The problem may be that the initial value isn't correctly set when the node is created.

Comment (ultijlam) [reply src/engine/runner.ts:188]: Yes, one way or another, there is a bug in the way the AST is evaluated, either in the way that results are stashed in the AST itself or improper aging of results.  Removing that one optimization definitely fixed the problem for now though.  This is a case where we need tests to document the expectation in issue #749 and verify that the right thing is being done by the addon.

Comment (Sidoine) [reply src/engine/runner.ts:188]: I fixed the bug in 04c102c842c24e056d68f014142c2a5ec5b57d43, you should be able to revert this code removal.

**Comment by ultijlam** [comment]
I can confirm that 04c102c842c24e056d68f014142c2a5ec5b57d43 fixed the issue with scripts not updating dynamically based on modifying script controls (checkboxes and lists).  I have made a separate pullup for the fix to #749.  I'm closing this pullup in favor of the new one.
